runner/android_test_orchestrator/java/androidx/test/orchestrator/AndroidTestOrchestrator.java

```diff
@@ -401,7 +403,15 @@ private String getOutputFile() {
     if (null == test) {
       return TEST_COLLECTION_FILENAME;
     } else {
-      return String.format(TEST_RUN_FILENAME, test);
+      // Max filename length is 255 characters, so tests with longer names have to be truncated to prevent crashes.
+      // To prevent collisions, we append a timestamp to the filename.
+      // Every file also has the dot and "txt" appended, so effectively the max length is 255 - 4 = 251 characters.
+
+      String millis = Long.toString(System.currentTimeMillis());
```

Comment:
Continuing the conversation from https://github.com/android/android-test/issues/1935#issuecomment-1745813824

If you use the timestamp, then every test run will have unique output, which means that it'll be very hard to compare for example two runs, because the filenames are not deterministic and don't match.

Reply:
agreed, will use hashCode

Reply:
done